### PR TITLE
CIRCSTORE-208: rename itemEffectiveLocationAtCheckOut

### PR DESCRIPTION
**Pull request by rjruizes**
https://issues.folio.org/browse/CIRCSTORE-208

Follow-up to [CIRCSTORE-157](https://github.com/folio-org/mod-circulation-storage/pull/194) which added a new field itemEffectiveLocationAtCheckOut. This field name was renamed in mod-circulation; this issue is to propagate that rename to mod-circulation-storage.

### CODE REVIEW

File: ramls/loan.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
       "description": "ID of the item lent to the patron",
       "type": "string"
     },
-    "itemEffectiveLocationAtCheckOut": {
+    "itemEffectiveLocationIdAtCheckOut": {
       "description": "The effective location, at the time of checkout, of the item loaned to the patron.",
       "type": "string",
       "pattern": "^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[1-5][a-fA-F0-9]{3}-[89abAB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$"
```
Comment (bohdan-suprun) [inline ramls/loan.json:27]:
```suggestion
      "$ref": "raml-util/schemas/uuid.schema"
```

File: ramls/loan-storage.raml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #%RAML 1.0
 title: Loan Storage
-version: v6.7
+version: v6.8
```
Comment (bohdan-suprun) [inline ramls/loan-storage.raml:3]: Please, update module descriptor API version as well.

File: ramls/examples/loan.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
   "itemId": "cb20f34f-b773-462f-a091-b233cc96b9e6",
   "loanDate": "2017-03-01T23:11:00-01:00",
   "dueDate": "2017-03-15T23:11:00-01:00",
-  "itemEffectiveLocationAtCheckOut": "07f83048-6737-4c6a-a3bc-c49cd9beaa38",
+  "itemEffectiveLocationIdAtCheckOut": "07f83048-6737-4c6a-a3bc-c49cd9beaa38",
```
Comment (marcjohnson-kint) [inline ramls/examples/loan.json:8]: This is a breaking compatibility change, the `loan-storage` interface will need changing to version `7.0` and all dependent modules will need updating for compatibility.

**Comment by marcjohnson-kint** [comment]
Thank you for updating this pull request to reflect the breaking change.

This change needs to be coordinated with changes to (at least) the following modules:
* ui-users
* mod-circulation
* stripes-core ([it appears](https://github.com/folio-org/stripes-core/blob/10bfbcf90c49cdb1244735147599f0313a784162/src/components/MainNav/AppList/tests/selectedApp.json#L14-L25))

There might also be other dependents like the API, regression or performance tests and reporting (which I am assuming you are aware of)

**Pull request merged**